chore: log to a blob that can be downloaded

devceline opened the pull request:
# Description

- Log to an array
- Allow the array to be downloaded as blob (file)
- Add UI button to download it
- Inject method to download it into `window` (just in case the UI is unresponsive or something)

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

chris13524 reviewed (changes requested):
This is awesome!!

Left some requests, also:
- Does the SDK support dumping logs from third-party apps or does it require this harness here? Goal would be to be able to dump logs from third-party apps using the SDK w/o them needing to implement stuff 
- Since the log dump may be truncated, should we prefix the log dump with some important "global" info such as client ID, token, any other state information?
- Are all the log entries prefixed with a timestamp?

devceline commented on src/utils/blobLogger.ts:
```diff
@@ -0,0 +1,52 @@
+  write(chunk: any): void {
+    this.logs.push(JSON.stringify(chunk))
+
+    // Using splice with a margin instead of unshift because logging
```
For a small number of logs (1000) I don't think it's worth the effort

devceline commented on src/utils/blobLogger.ts:
```diff
@@ -0,0 +1,52 @@
+export class ArrayLogStream {
+  private logs: string[]
+  private MAX_LOG_SIZE = 200
```
Ya I was being a bit stingy there, raised it to 1k

devceline commented:
> This is awesome!!
> 
> Left some requests, also:
> 
> * Does the SDK support dumping logs from third-party apps or does it require this harness here? Goal would be to be able to dump logs from third-party apps using the SDK w/o them needing to implement stuff.
>   
>   * Next we should pull this SDK version into GM dApp and just have it work. And a similar approach for Web3Modal/sign client/etc.
> * Since the log dump may be truncated, should we prefix the log dump with some important "global" info such as client ID, token, any other state information?
> * Are all the log entries prefixed with a timestamp?

1. The SDK allows them to pass a custom logger. The reason I did not include this in the docs is depending on their environment this might not work. Eg, RN vs Browser environments or if for any reason the SDK was used on node.
2. Added
3. Added

devceline commented:
Closing in favor of solution in JS core.
